drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1236,6 +1238,13 @@ static void usbd_work_handler(struct k_work *item)
 
 	while ((ev = usbd_evt_get()) != NULL) {
 
+		if (!dev_ready()) {
+			if (ev->evt_type != USBD_EVT_POWER) {
```

Comment:
Could do `if (!dev_ready() && ev->evt_type != USBD_EVT_POWER)` to shave some lines.